packages/rs-drive-abci/src/execution/engine/run_block_proposal/v0/mod.rs

```diff
@@ -68,6 +68,11 @@ where
         platform_version: &PlatformVersion,
     ) -> Result<ValidationResult<block_execution_outcome::v0::BlockExecutionOutcome, Error>, Error>
     {
+        // Cleanup block cache before we execute a new proposal
+        let mut drive_cache = self.drive.cache.write().unwrap();
+        drive_cache.cached_contracts.clear_block_cache();
+        drop(drive_cache);
+
```

Comment:
This should be moved down a bit in the function.

Reply:
How far down you want to have it? It should be before we actually using the cache, because previous execution might fail and we could help something there from the previous run.

Reply:
It needs to be after that we make sure we are in the correct block proposal.